lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2649,6 +2649,114 @@ def calc_arrow(uvw, angle=15):
 
     quiver3D = quiver
 
+    def voxels(self, filled, color=None):
+        """
+        Plot a set of filled voxels
+
+        All voxels are plotted as 1x1x1 cubes on the axis, with filled[0,0,0]
+        placed with its lower corner at the origin. Occluded faces are not plotted
+
+        Parameters
+        ----------
+        filled : np.array
+            A 3d array of values, with truthy values indicating which voxels
+            to fill
+
+        color : array_like
+            Either a single value or an array the same shape as filled,
+            indicating what color to draw the faces of the voxels. If None,
+            plot all voxels in the same color, the next in the color sequence
+
+        Returns
+        -------
+        list of `Poly3DCollection`
+            The square faces which were plotted, in an arbitrary order
```

Comment:
Each element of the returned list is a single face.

> Also, "arbitrary order" is too ambiguous.

The order is `[XY-oriented] + [YZ-orientated] + [ZX-orientated]`, where each of those is in a lexicographic ordering over some permutation of their position.

The goal here was just to return a set of objects that borders etc could be put on, so the ordering is not documented.

I suppose in principle the return value could be a dict indexed by coordinate, containing a single `Poly3DCollection` for all the faces rendered for that voxel, but that would be more work